From 226606ec42fb0b5fb0efd13c2401a5418434efe9 Mon Sep 17 00:00:00 2001
From: Ninja Tekkaman <tekkamanninja@gmail.com>
Date: Fri, 7 Aug 2009 08:04:52 -0400
Subject: [PATCH 162/247] drivers: usb: gadget: call switch_dev_unregister in mass storage unbind callback

This fixes a problem unloading the android gadget driver when built as a module

Signed-off-by: Mike Lockwood <lockwood@android.com>
---
 drivers/usb/gadget/f_mass_storage.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/gadget/f_mass_storage.c b/drivers/usb/gadget/f_mass_storage.c
index 9f3dd91..5e9e2a0 100644
--- a/drivers/usb/gadget/f_mass_storage.c
+++ b/drivers/usb/gadget/f_mass_storage.c
@@ -2684,6 +2684,7 @@ fsg_function_unbind(struct usb_configuration *c, struct usb_function *f)
 	/* Free the data buffers */
 	for (i = 0; i < NUM_BUFFERS; ++i)
 		kfree(fsg->buffhds[i].buf);
+	switch_dev_unregister(&fsg->sdev);
 }
 
 static int __init
-- 
1.7.0.4

